src/sonic-config-engine/minigraph.py

```diff
@@ -600,34 +648,47 @@ def parse_xml(filename, platform=None, port_config_file=None):
     bgp_peers_with_range = None
     deployment_id = None
 
-    hwsku_qn = QName(ns, "HwSku")
+    
+    #hostname is the asic_name, get the asic_id from the asic_name
+    if hostname is not None:
+        asic_id = hostname[-1]
```

Comment:
is it always true that asic id will be single digit? can you use regex to match the id?

Reply:
The assumption was the asic_id will be single digit. However as per the comment add a new function to use regex to match the id.